### PR TITLE
[LW-32] Add Exception module (#57)

### LINKED ISSUE #57: Add 'logException' function and utilities
It would be really good to have some utility function to log exceptions with `Error` severity, something like that:

```haskell
logException :: (WithLogger m, Exception e) => e -> m ()
```

I would also appreciate to have function like `catch` which calls `logException` of some `IO` action if this exception is thrown.

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #57

### CODE REVIEW

File: src/System/Wlog/Exception.hs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
```
Comment (chshersh) [inline src/System/Wlog/Exception.hs:1]: Can you research and say why do you need `AllowAmbiguousTypes` in this module?

File: src/System/Wlog/Exception.hs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+       ) where
+
+import Universum
+
+import Control.Exception.Base (Exception)
+
+import System.Wlog.CanLog (WithLogger, WithLoggerIO, logError)
+
+-- | Logs exception's description with ''System.Wlog.Severity.Error' 'System.Wlog.Severity.Severity'
+logException :: (WithLogger m, Exception e) => e -> m ()
```
Comment (chshersh) [inline src/System/Wlog/Exception.hs]: I think, you need to specify `e` as a first type variable with `forall`. I expect that in future users might want to use `-XTypeApplications` to specify exception to log.

File: src/System/Wlog/Exception.hs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+logException e = logError $ show e
```
Comment (chshersh) [inline src/System/Wlog/Exception.hs]: `logException = logError . show` :)

File: src/System/Wlog/Exception.hs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+-- | Runs the action, if an exception is raised the 'logException' is executed.
+catchLog :: forall m e . (WithLoggerIO m, MonadCatch m, Exception e) => m () -> m ()
```
Comment (chshersh) [inline src/System/Wlog/Exception.hs]: Same here: swap `m` and `e`.

File: src/System/Wlog/Exception.hs (hunk 1 above)
Comment (chshersh) [reply src/System/Wlog/Exception.hs:1]: Nevermind, I got it :slightly_smiling_face:

**Review by chshersh — CHANGES_REQUESTED** [review]

**Comment by chshersh** [comment]
Also, I would to see examples of usages of these functions. It might be not obvious how to use then, because you need to specify type of exception using `-XTypeApplications`. Adding examples to documentation with `>>>` is a good idea!

File: src/System/Wlog/Exception.hs (hunk 2 above)
Comment (vrom911) [reply src/System/Wlog/Exception.hs]: Yes, true! This definitely should be done in right order..

File: src/System/Wlog/Exception.hs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+runWithExceptionLog = catchLog @IOException (liftIO simpleIOfun)
+
+simpleIOfun :: IO ()
+simpleIOfun = getLine >>= readFile >>= putStrLn
+
+@
+
+and when run you will get:
+
+>>> stack exec except-log
```
Comment (chshersh) [inline src/System/Wlog/Exception.hs]: You're running this example with `stack exec except-log` command, but there's no `except-log` target in this project. You can either add this target to project or use some abstract generic command in CLI like `run-main-from-this-example`.

**Review by chshersh — CHANGES_REQUESTED** [review]

**Review by chshersh — APPROVED** [review]

**Pull request merged**